src/types/Uint16Array.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/// @dev An uint16 array of max 16 values packed into a single uint256
+type Uint16Array is uint256;
+using Uint16ArrayLibrary for Uint16Array global;
+error IndexOutOfBounds();
+error InvalidArrLength();
+
+function toUint256(uint16[] memory inputArray) pure returns (uint256 uint16Array) {
+    return Uint16Array.unwrap(toUint16Array(inputArray));
+}
+
+function fromUnderlying(uint256 value) pure returns (Uint16Array) {
+    return Uint16Array.wrap(value);
+}
+
+// Helper for creating a packed uint256 from a uint16 array
+function toUint16Array(uint16[] memory inputArray) pure returns (Uint16Array uint16Array) {
```

Comment:
imo better to keep these in the library, global functions are a bit awkward and not standard for uniswap style. Generally the `using for` functions at the top of the library and any extraneous helpers at the bottom